src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -829,6 +829,50 @@ public static void DeserializeUserDefinedDictionaryThrows()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Parse<IImmutableDictionaryWrapper>(json));
         }
 
+        [Fact]
+        public static void Regression38643_Serialize()
+        {
+            // Arrange
+            var value = new Regression38643_Parent()
+            {
+                Child = new Dictionary<string, Regression38643_Child>()
+                {
+                    ["1"] = new Regression38643_Child()
+                    {
+                        A = "1",
+                        B = string.Empty,
+                        C = Array.Empty<string>(),
+                        D = Array.Empty<string>(),
+                        F = Array.Empty<string>(),
+                        K = Array.Empty<string>(),
+                    }
+                }
+            };
+
+            var actual = JsonSerializer.ToString(value, new JsonSerializerOptions() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+            Console.WriteLine(actual);
+
+            // Assert
+            Assert.NotNull(actual);
+            Assert.NotEmpty(actual);
+        }
+
+        [Fact]
+        public static void Regression38643_Deserialize()
+        {
+            // Arrange
+            string json = "{\"child\":{\"1\":{\"a\":\"1\",\"b\":\"\",\"c\":[],\"d\":[],\"e\":null,\"f\":[],\"g\":null,\"h\":null,\"i\":null,\"j\":null,\"k\":[]}}}";
+
+            var actual = JsonSerializer.Parse<Regression38643_Parent>(json, new JsonSerializerOptions() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
```

Comment:
@steveharter, just want to confirm. If the user says `CamelCase`, we expect it to fail if they provide something that matches the `ExactCase`, correct?

For example, here the `Regression38643_Parent.Child` property is cased as `Child`. We would expect `actual.Child` to be null if the JSON payload cased it as `Child` as well (rather than as camelCase `child`), correct?

Do we also expect it to be `null` or to `fail`? It currently appears to set it to `null` and you don't get any error on there being an unused entry.

Reply:
> If the user says CamelCase, we expect it to fail if they provide something that matches the ExactCase, correct?

Yes we should fail. The user would have to enable case-insensitivity if they want that.

Reply:
> Yes we should fail.

Under what conditions do we want the failure to occur?

Currently, it is just setting `Child` to `null` (since the payload doesn't have a matching entry named `child`) and appears to be ignoring the json payload entry titled `Child`.